### PR TITLE
mon/scrub: add mon scrub latency

### PR DESCRIPTION
**Pull request by YiteGu**
If mon store.db very big, scrub will be a very
time-consuming operation.

Fixes: https://tracker.ceph.com/issues/66873

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by YiteGu** [comment]
Hi @ronen-fr take a look this PR

**Review by ronen-fr — CHANGES_REQUESTED** [review]
The tracker - an enhancement request - is devoid of any content.
Please extend it to explain what the problem is, and what is the change requested.

**Comment by YiteGu** [comment]
> The tracker - an enhancement request - is devoid of any content. Please extend it to explain what the problem is, and what is the change requested.

thanks @ronen-fr, updated tracker.

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -5661,10 +5661,13 @@ void Monitor::handle_scrub(MonOpRequestRef op)
       if (scrub_result.size() == quorum.size()) {
         scrub_check_results();
         scrub_result.clear();
-        if (scrub_state->finished)
+        if (scrub_state->finished) {
+          const utime_t lat = ceph_clock_now() - scrub_state->start;
+          dout(10) << __func__ << " mon scrub latency: " << lat << dendl;
```
Comment (ronen-fr) [inline src/mon/Monitor.cc:5666]: This is not a "monitor scrub", and it's not a 'mon scrub latency'.

**Review by ronen-fr — DISMISSED** [review]
I do not see the point in this PR.
Can you point to customers' request for this feature?
The cluster log already shows the times for scrub starts and ends,
and the 'pg dump' command shows the last scrub duration.

**Comment by YiteGu** [comment]
> I do not see the point in this PR. Can you point to customers' request for this feature? The cluster log already shows the times for scrub starts and ends, and the 'pg dump' command shows the last scrub duration.

Hi, Ronen
mon will do scrub everyday, controlled by config `mon_scrub_interval` , this PR can help us known mon scrub latency, not pg scrub. If you feel not need, I close it.

**Comment by ronen-fr** [comment]
> Hi, Ronen
> mon will do scrub everyday, controlled by config `mon_scrub_interval` , this PR can help us known mon scrub latency, not pg scrub. If you feel not need, I close it. I am writing this PR because this case https://tracker.ceph.com/issues/64824. I want to express more about auxiliary enhancement.

I may be misunderstanding something. I'll ask @rzarzynski to take a look de-novo

**Review by rzarzynski — APPROVED** [review]

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrackercephcomissues68445

**Pull request merged**